### DISCUSSION TITLE
Multi step forms

### DISCUSSION THREAD

**Discussion by jlgea** [myndpm/open-source#35] [Se habla Español]
Hola Mateo! Estoy retomando mi proyecto y quisiera saber si se pueden crear los formularios tipo "multi step".

Gracias mil!

**Comment by matheo** [comment]
Hola Jose Luis!
Es posible implementarlos y podríamos hacerlo con el `mat-stepper` de Angular Material.

Me pregunto cómo quieres implementarlo a nivel de backend?
guardando cada paso diligenciado o sólo cuando todo el form se complete?

Yo hice una implementación con múltiples rutas: https://mynd.dev/demos/dyn-forms/stepper-form/step-1
pero eso requeriría tambien unos Guards para no permitir que accedan a pasos futuros con la URL

**Reply by jlgea** [reply]
Sí. En función de las respuestas, el backend enviaría el siguiente formulario hasta completar todos los formularios requeridos.

Podría usar las condiciones que implementas en la librería, pero prefiero hacerlo desde el backend porque así, el día de mañana puedo cambiar el frontend por cualquier otro. ¿Tú que opinas?

**Reply by matheo** [reply]
Suena bien, pero la cantidad de pasos variaría o serían fijos?
Necesito saber más requerimientos específicos para poder visualizar qué solución se ajusta más.

El `mat-stepper` lo veo aplicable para cuando la cantidad de pasos es fijo,
o un control personalizado de no ser así. Has hecho un boceto de lo que quieres? o el diseño?
